channeldb/mp_payment.go

```diff
@@ -130,7 +164,20 @@ func serializeHTLCFailInfo(w io.Writer, f *HTLCFailInfo) error {
 		return err
 	}
 
-	return nil
+	// Write failure. If there is no failure message, write an empty
+	// byte slice.
+	var messageBytes bytes.Buffer
+	if f.Message != nil {
```

Comment:
encoding should follow the struct order IMO

Reply:
struct order updated

Reply:
Because it doesn't matter in which order the fields are stored. What do you want me to do?